src/update_version.rs

```diff
@@ -0,0 +1,61 @@
+use library::{self, Function, Type};
+use Library;
+
+fn update_function_version(functions: &mut Vec<Function>, lib: *const Library) {
+    for function in functions {
+        let mut current_version = None;
+        for parameter in &function.parameters {
+            if let Some(new_version) = match (current_version,
+                                              match *unsafe { (*lib).type_(parameter.typ) } {
```

Comment:
The result is a bit ugly by doing this way. I think it's preferable to keep it it as is.